package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,371 @@
+does not represent a sample from equilibrium. This violates a basic
+assumption of the anisotropic kernel. The anisotropic kernel is created to act
+as a discrete approximation for the Fokker-Plank equation from statistical
+mechanics, results from a non-equilibrium trajectory shouldn't be interpreted
+for this reason. This tutorial shows how to use the diffusionmap function.
+First load all modules and test data ::
+
+   >>> import MDAnalysis
+   >>> import numpy as np
+   >>> import MDAnalysis.analysis.diffusionmap as diffusionmap
+   >>> from MDAnalysis.tests.datafiles import PSF,DCD
+
+Given a universe or atom group, we can calculate the diffusion map from
+that trajectory using :class:`DiffusionMap`:: and get the corresponding
+eigenvalues and eigenvectors.
+
+   >>> u = MDAnalysis.Universe(PSF,DCD)
+
+We leave determination of the appropriate scale parameter epsilon to the user,
+[Clementi1]_ uses a complex method involving the k-nearest-neighbors of a
+trajectory frame, whereas others simple use a trial-and-error approach with
+a constant epsilon. Users wishing to use a complex method to determine epsilon
+will have to initialize a distance matrix and scale it appropriately themselves
+and then pass the new scaled distance matrix as a parameter.
+
+   >>> dmap = diffusionmap.DiffusionMap(dist_matrix, epsilon =2)
+   >>> dmap.run()
+   >>> eigenvalues = dmap.eigenvalues
+   >>> eigenvectors = dmap.eigenvectors
+
+From here we can perform an embedding onto the k dominant eigenvectors. This
+is similar to the idea of a transform in Principal Component Analysis, but the
+non-linearity of the map means there is no explicit relationship between the
+lower dimensional space and our original trajectory. However, this is an
+isometry (distance preserving map), which means that points close in the lower
+dimensional space are close in the higher-dimensional space and vice versa.
+In order to embed into the most relevant low-dimensional space, there should
+exist some number of dominant eigenvectors, whose corresponding eigenvalues
+diminish at a constant rate until falling off, this is referred to as a
+spectral gap and should be somewhat apparent for a system at equilibrium with a
+high number of frames.
+
+   >>> num_eigenvectors = # some number less than the number of frames-1
+   >>> dmap.transform(num_eigenvectors)
+
+From here it can be difficult to interpret the data, and is left as a task
+for the user. The `diffusion distance` between frames i and j is best
+approximated by the euclidean distance  between rows i and j of
+self.diffusion_space. A Jupyter notebook providing an analysis of protein
+opening and closing is provided [here](#TODO url to notebook)
+
+Classes
+-------
+
+.. autoclass:: DiffusionMap
+.. autoclass:: DistMatrix
+
+References
+---------
+
+If you use this Dimension Reduction method in a publication, please
+reference:
+..[Ferguson1] Ferguson, A. L.; Panagiotopoulos, A. Z.; Kevrekidis, I. G.;
```

Comment:
Nonlinear dimensionality reduction in molecular simulation: The diffusion map approach